simulator/modules/fetch/fetch.cpp

```diff
@@ -188,6 +200,30 @@ void Fetch<FuncInstr>::clock( Cycle cycle)
 
     /* sending to decode */
     wp_datapath->write( std::move( instr), cycle);
+
+    if ( prefetch_enabled && !is_wrong_path)  // prefetch next line if enabled and wrong path isn't used
+    {
+        prefetch_next_line(target.address);
+    }
+}
+
+template<typename FuncInstr>
+void Fetch<FuncInstr>::prefetch_next_line(Addr requested_addr)
+{
+    uint32 line_size = uint32(config::instruction_cache_line_size); // line size
+    size_t line_bits = std::countr_zero(line_size); // number of offset bits
+
+    Addr addr_mask = bitmask<Addr>( line_bits); // bit mask to extract the offset value
+    Addr offset = requested_addr & addr_mask; // offset
+
+    Addr next_line_addr = requested_addr + line_size; // the address that will be on the next line
+    uint32 fetchahead_distance = fetchahead_size; // setting the fetchahead
+
+   /* if the fetchahead is reached and there is no line in the cache, write the line to the cache */
+   if ( offset >= (line_size - fetchahead_distance) && !tags->lookup( next_line_addr))
+   {
```

Comment:
Our code guideline is to use as less brackets as possible: https://github.com/MIPT-ILab/mipt-mips/wiki/Code-style-guidelines#pythonesque-statements